File: src/main/java/pe/nanamochi/banchus/entities/commons/ScoringType.java
package pe.nanamochi.banchus.entities.commons;

import lombok.AllArgsConstructor;
import lombok.Getter;

@Getter
@AllArgsConstructor
public enum ScoringType {
  SCORE(0),
  ACCURACY(1),
  COMBO(2),
  SCORE_V2(3);

  private final int value;

  public static ScoringType fromValue(int value) {
    for (ScoringType type : values()) {
      if (type.value == value) {
        return type;
      }
    }
    return SCORE;
  }
}

File: banchus-packet/src/main/kotlin/pe/nanamochi/banchus/components/ScoringType.kt
package pe.nanamochi.banchus.components

enum class ScoringType(val value: Int) {
    SCORE(0),
    ACCURACY(1),
    COMBO(2),
    SCORE_V2(3);

    companion object {
        fun fromValue(value: Int): ScoringType = entries.find { it.value == value } ?: SCORE
    }
}
